iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/SortedArgumentListCheck.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.checks;
+
+import java.util.List;
+import java.util.Set;
+import org.sonar.check.Rule;
+import org.sonar.iac.common.api.checks.CheckContext;
+import org.sonar.iac.common.api.checks.IacCheck;
+import org.sonar.iac.common.api.checks.InitContext;
+import org.sonar.iac.docker.checks.utils.CommandDetector;
+import org.sonar.iac.docker.symbols.ArgumentResolution;
+import org.sonar.iac.docker.tree.api.RunInstruction;
+
+@Rule(key = "S7018")
+public class SortedArgumentListCheck implements IacCheck {
+  private static final String MESSAGE = "Sort these package names alphanumerically.";
+  private static final String INSTALL_COMMAND = "install";
+  private static final String ADD_COMMAND = "add";
+  private static final int MINIMAL_LENGTH_TO_SORT = 3;
+
+  private static final CommandDetector DEBIAN_PACKAGE_MANAGER_DETECTOR = CommandDetector.builder()
+    .with(Set.of("apt", "apt-get", "aptitude"))
+    .withAnyFlag()
+    .with(INSTALL_COMMAND)
+    .withAnyFlag()
+    .withOptionalRepeatingExcept("EOF")
```

Comment:
I have bad news for you :sweat_smile: `EOF` is the token we typically use in our examples of the here-doc form, but in fact the user can specify [any token they want](https://docs.docker.com/reference/dockerfile/#here-documents):
```
COPY <<-EOT /script.sh
  echo "hello ${FOO}"
EOT
```
I guess instead, we should match until we have any multi-command operator (`&&,` `||,` `|,`...).
It may be something already handled by `CommandDetector` API.

Reply:
Yes, this matcher was only for the case of Heredoc. I think I'll disable the rule for heredocs completely, because with all commands glued together sorting can lead to unexpected results.